src/tlv/tlv_library.h

```diff
@@ -0,0 +1,283 @@
+    enum {                                                                               \
+        TAG_LIST(__X_DEFINE_TLV__TAG_FLAG)                                               \
+    };                                                                                   \
+                                                                                         \
+    /* A dynamically generated function that maps tags to flags (for internal use). */   \
+    /* */                                                                                \
+    /* @param[in] tag The tag to get the flag of*/                                       \
+    /* @return The associated flag or 0 if none exist */                                 \
+    static inline TLV_flag_t PARSE_FUNCTION_NAME##_tag_to_flag(TLV_tag_t tag) {          \
+        switch (tag) {                                                                   \
+            TAG_LIST(__X_DEFINE_TLV__TAG_TO_FLAG_CASE)                                   \
+            default:                                                                     \
+                return 0;                                                                \
+        }                                                                                \
+    }                                                                                    \
+                                                                                         \
+    /* A dynamically generated TLV parser function for this TLV use case. */             \
+    /* */                                                                                \
+    /* Parses a TLV payload using the tag and handlers of TAG_LIST */                    \
+    /* */                                                                                \
+    /* @param[in] payload the TLV payload to parse */                                    \
+    /* @param[out] tlv_out The output data of this parser, written in by the handlers */ \
+    /* @param[out] received_tags_flags The tag reception structure */                    \
+    /* @return whether it was successful */                                              \
+    static inline bool PARSE_FUNCTION_NAME(const buffer_t *payload,                      \
+                                           void *tlv_out,                                \
+                                           TLV_reception_t *received_tags_flags) {       \
+        /* Create a local array of handlers to give to the generic parser */             \
+        _internal_tlv_handler_t handlers[PARSE_FUNCTION_NAME##_TAG_COUNT] = {            \
+            TAG_LIST(__X_DEFINE_TLV__TAG_CALLBACKS)                                      \
+        };                                                                               \
+        return _parse_tlv_internal(handlers,                                             \
```

Comment:
Rust SDK will have to call this function directly

Reply:
Handled through use case API